src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -362,80 +428,267 @@ private void ToggleGesture(RoutedEvent routedEvent)
+					// even if the AllowDrop flag has not been set, we have to update the IsDragOver state.
+					SetIsDragOver(pt, true);
+
+					if (!AllowDrop // The Drag and Drop "routed" events are raised only on controls that opted-in
+						|| wasDragOver) // If we already had a DragEnter do not raise it twice
+					{
+						bubblingMode = BubblingMode.IgnoreElement;
+					}
 					break;
-				case RoutedEventFlag.PointerExited:
-					OnManagedPointerExited(ptArgs);
+				}
+
+				case RoutedEventFlag.DragOver:
+					// As the IsDragOver is expected to reflect the state of the current element **and the state of its children**,
+					// even if the AllowDrop flag has not been set, we have to update the IsDragOver state.
+					SetIsDragOver(((global::Windows.UI.Xaml.DragEventArgs)args).SourceId, true);
+
+					if (!AllowDrop) // The Drag and Drop "routed" events are raised only on controls that opted-in
+					{
+						bubblingMode = BubblingMode.IgnoreElement;
+					}
 					break;
-				case RoutedEventFlag.PointerCanceled:
-					OnManagedPointerCancel(ptArgs);
+
+				case RoutedEventFlag.DragLeave:
+				case RoutedEventFlag.Drop:
+				{
+					var pt = ((global::Windows.UI.Xaml.DragEventArgs)args).SourceId;
+					var wasDragOver = IsDragOver(pt);
+
+					// As the IsDragOver is expected to reflect the state of the current element **and the state of its children**,
+					// even if the AllowDrop flag has not been set, we have to update the IsDragOver state.
+					SetIsDragOver(pt, false);
+
+					if (!AllowDrop // The Drag and Drop "routed" events are raised only on controls that opted-in
+						|| !wasDragOver) // No Leave or Drop if we was not effectively over ^^
+					{
+						bubblingMode = BubblingMode.IgnoreElement;
+					}
 					break;
-				// No local state (over/pressed/manipulation/gestures) to update for
-				//	- PointerCaptureLost:
-				//	- PointerWheelChanged:
+				}
 			}
 		}
 
-		partial void PrepareManagedManipulationEventBubbling(RoutedEvent routedEvent, ref RoutedEventArgs args, ref bool isBubblingAllowed)
+		[ThreadStatic]
+		private static uint _lastDragStartFrameId;
+
+		private void OnDragStarting(DraggingEventArgs args)
 		{
-			// When we bubble a manipulation event from a child, we make sure to abort any pending gesture/manipulation on the current element
-			if (routedEvent != ManipulationStartingEvent && _gestures.IsValueCreated)
+			if (args.DraggingState != DraggingState.Started // This UIElement is actually interested only by the starting
+				|| !CanDrag // Sanity ... should never happen!
+				|| !args.Pointer.Properties.IsLeftButtonPressed
+
+				// As the pointer args are always bubbling (for to properly update pressed/over state and manipulations),
+				// if a parent is CanDrag == true, its gesture recognizer might (should) also trigger the DragStarting.
+				// But: (1.) on UWP only the top-most draggable element starts the drag operation;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6334034)